src/main/java/gov/usgs/wma/waterdata/data/DataController.java

```diff
@@ -59,14 +59,22 @@ public void getTimeSeries(
 					@ExampleObject(name = "waterML", value = "WaterML", description = "Water ML") }) @RequestParam(value = "f", required = false, defaultValue = "waterml") String mimeType,
 			HttpServletResponse response) throws HttpMediaTypeNotAcceptableException, IOException {
 
-		// not using the ContentType returned yet, since we only have waterML
-		determineContentType(mimeType, List.of(ContentType.waterml));
+		ContentType contentType = determineContentType(mimeType, List.of(ContentType.json, ContentType.waterml));
 		String rtn = null;
 		String bestTS = best == null ? CollectionParams.PARAM_MATCH_ANY : best.toString().toLowerCase();
 
+		if (contentType.isJson() && !bestTS.equals("true")) {
+			throw new HttpMediaTypeNotAcceptableException("Json content is only available with parameter best=true");
+		}
+
 		if (Domain.includesGroundWaterLevels(domains) && dataType.isStatisticalTimeSeries()) {
-			rtn = timeSeriesDao.getTimeSeriesWaterML(monLocIdentifier, bestTS);
-			response.setContentType(MediaType.APPLICATION_XML_VALUE);
+			if(contentType.isJson()) {
+				rtn = timeSeriesDao.getTimeSeries(monLocIdentifier, bestTS);
+				response.setContentType(MediaType.APPLICATION_JSON_VALUE);
+			} else {
+				rtn = timeSeriesDao.getTimeSeriesWaterML(monLocIdentifier, bestTS);
+				response.setContentType(MediaType.APPLICATION_XML_VALUE);
+			}
```

Comment:
If we ever had more content types then this should be polymorphic. Say a enum that contained the rules or linked to a class that contained the rules. Long if-else chains are hard to follow.

Reply:
Totally agree that long if chains are hard to follow, essentially when the code blocks are dozens/hundreds of lines long.
In this case, it is hard to say how many more content type there will be. There will be csv at some point.
This should be part of the refactor to using the accept headers for the content type.